src/internal/packager/images/pull.go

```diff
@@ -27,329 +32,329 @@ import (
 	"github.com/google/go-containerregistry/pkg/v1/empty"
 	clayout "github.com/google/go-containerregistry/pkg/v1/layout"
 	"github.com/google/go-containerregistry/pkg/v1/partial"
-	ctypes "github.com/google/go-containerregistry/pkg/v1/types"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/google/go-containerregistry/pkg/v1/types"
 	"github.com/moby/moby/client"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"golang.org/x/sync/errgroup"
 )
 
 // ImgInfo wraps references/information about an image
 type ImgInfo struct {
-	RefInfo        transform.Image
-	Img            v1.Image
-	HasImageLayers bool
+	RefInfo transform.Image
+	Img     v1.Image
 }
 
-// PullAll pulls all of the images in the provided tag map.
-func (i *ImageConfig) PullAll() ([]ImgInfo, error) {
-	var (
-		longer            string
-		imageCount        = len(i.ImageList)
-		refInfoToImage    = map[transform.Image]v1.Image{}
-		referenceToDigest = make(map[string]string)
-		imgInfoList       []ImgInfo
-	)
-
-	type digestInfo struct {
-		refInfo transform.Image
-		digest  string
-	}
-
+// Pull pulls all of the images from the given config.
+func Pull(ctx context.Context, cancel context.CancelFunc, cfg PullConfig) (map[transform.Image]v1.Image, error) {
```

Comment:
It would be more idiomatic to only pass the context as an argument and then define a new sub context with cancel in the beginning of this function.

Reply:
Refactored to not need `cancel` at all as context usage is is now more properly scoped